source_integration_test.go

```diff
@@ -91,3 +92,92 @@ func TestSource_ConsistentSnapshot(t *testing.T) {
 	testutils.ReadAndAssertInsert(ctx, is, sourceIterator, user6)
 	testutils.ReadAndAssertDelete(ctx, is, sourceIterator, user4)
 }
+
+func TestSource_MultipleSnapshotFetches(t *testing.T) {
+	ctx := testutils.TestContext(t)
+	is := is.New(t)
+
+	db := testutils.Connection(t)
+
+	userTable.Recreate(is, db)
+
+	// insert 100 users, delete first 20 so that the min primary key code path
+	// is hit
```

Comment:
Or maybe delete the first 10, and 10 in the middle, and try (re)starting at different points.

Reply:
Hello @hariso, I've been going on for a while with this test, and I believe I just didn't really document it well why was there. It is ensuring proper iteration, ensuring this [bug](https://github.com/conduitio-labs/conduit-connector-mysql/issues/52#issuecomment-2381625825) doesn't happen.